### PR TITLE
Customer registration, login, forgotpassword

### PR DESCRIPTION
**Pull request by vamcart**
#6

### CODE REVIEW

**Comment by vamcart** [comment]
And here is the my working copy of this changes.
All works fine for me.
https://cezerin2.slack.com/files/UFVAW9HA4/FGCL59ETV/cezerin-customer-register-login-account.zip

**Comment by vamcart** [comment]
To test new email templates, features, i have local smtp server.
Setup your smtp server at /config/server.js to test account activation, password recovery feature.

New email templates and options added to the "npm run setup"
/src/api/server/setup.js

**Comment by vamcart** [comment]
If you have error when you click register button, try to clear cache in Chrome - Developer Console - Application - Clear storage - Clear site data.

actions.js:774 Uncaught (in promise) TypeError: Cannot read property 'retrieve' of undefined
in /src/store/shared/actions.js
:
export const registerUser = (data, callback) => async (dispatch, getState) => {
    const response = await api.ajax.register.retrieve(data);
    dispatch(handleRegisterProperties(response.json));
};
this string:
api.ajax.register.retrieve(data);

**Comment by LuBuss** [comment]
Add a few missing files.
- you can create an account from the checkout

Work is still required in:
- Customer Login (have not tested)
- Customer Register (have not tested)

**Comment by LuBuss** [comment]
I Verify all working

**Comment by vamcart** [comment]
ok, merge this pr.

> I Verify all working

**Comment by LuBuss** [comment]
Only verified working, but have not checked code.

File: src/server/index.js [hunk 1]
```diff
@@ -13,17 +13,28 @@ import pageRendering from './pageRendering';
...
 
+const ADMIN_INDEX_PATH = path.resolve('public/admin/index.html');
 const STATIC_OPTIONS = {
 	maxAge: 31536000000 // One year
 };
 
 app.set('trust proxy', 1);
 app.use(helmet());
-
+app.get('/images/:entity/:id/:size/:filename', (req, res, next) => {
```
Comment (ngoyal16) [inline src/server/index.js]: this code should be remove as it is part of api now.

File: src/server/index.js [hunk 2]
```diff
@@ -13,17 +13,28 @@ import pageRendering from './pageRendering';
...
+	// A stub of image resizing (can be done with Nginx)
+	const newUrl = `/images/${req.params.entity}/${req.params.id}/${
+		req.params.filename
+	}`;
+	req.url = newUrl;
+	next();
+});
 app.use(express.static('public/content', STATIC_OPTIONS));
 app.use('/assets', express.static('theme/assets', STATIC_OPTIONS));
+app.use('/admin-assets', express.static('public/admin-assets', STATIC_OPTIONS));
```
Comment (ngoyal16) [inline src/server/index.js]: remove admin code from this. as we have an admin as a separate project

File: src/server/index.js [hunk 3]
```diff
@@ -13,17 +13,28 @@ import pageRendering from './pageRendering';
...
 app.use('/sw.js', express.static('theme/assets/sw.js'));
-
+app.use('/admin', (req, res) => {
```
Comment (ngoyal16) [inline src/server/index.js]: remove admin code from this. as we have an admin as a separate project

**Review by ngoyal16 — CHANGES_REQUESTED** [review]

**Comment by vamcart** [comment]
When this PR will be merged?!

I test it on my local machine and it's ok.

New version of cezerin2-client - ok.
Customer Registration - ok.
Customer Activation - ok.
Customer Order Checkout - ok.
Customer Logoff - ok.
Customer Account Page - ok.
Customer Edit Address Form - ok.
New settings for "npm run setup" - ok.

All of this works fine at my local machine.

**Review by LuBuss — APPROVED** [review]

**Pull request merged**